experimental/Schemes/SpaceGerms.jl

```diff
@@ -0,0 +1,233 @@
+  R = base_ring(base_ring(I))
+  R == ambient_ring(X) || error("rings are not compatible")
+  J = ideal(R, [numerator(p) for p in gens(I)]) # expected to be a single k-point, only tested in nex line
+  a = _maxideal_to_point(J)
+  Y = SpaceGerm(X,a)
+  return Y
+end
+
+function SpaceGerm(X::Spec, I::MPolyQuoLocalizedIdeal)
+  L = base_ring(I)
+  R = base_ring(L)
+  R == ambient_ring(X) || error("rings are not compatible")
+  J = ideal(R, [numerator(p) for p in lift.(gens(I))]) + modulus(quotient_ring(L))
+  a = _maxideal_to_point(J)
+  Y = SpaceGerm(X,a)
+  return Y
+end  
+
+function SpaceGerm(X::Spec, I::MPolyIdeal)
+  R = base_ring(I)
+  R == ambient_ring(X) || error("rings are not compatible")
+  a = _maxideal_to_point(I)
+  Y = SpaceGerm(X,a)
+  return Y
+end
+
+function SpaceGerm(X::Spec, I::MPolyQuoIdeal)
+  A = base_ring(I)
+  A == OO(X) || error("rings are not compatible")
+  R = base_ring(A)
+  I = ideal(R, lift.(gens(I))) + modulus(A)
+  a = _maxideal_to_point(I)
+  Y = SpaceGerm(X,a)
+  return Y
+end
+
+function germ_at_point(X::Spec, I::Ideal)
+  Y = SpaceGerm(X, I)
+  restr_map = SpecMor(Y, X, hom(OO(X), OO(Y), gens(OO(Y)), check=false), check=false)
+  return Y, restr_map
+end
+
+function germ_at_point(X::Spec, a::Vector)
+  Y = SpaceGerm(X, a)
+  restr_map = SpecMor(Y, X, hom(OO(X), OO(Y), gens(OO(Y)), check=false), check=false)
+  return Y, restr_map
+end
+
+function germ_at_point(A::MPolyRing, I::Ideal)
+  X = Spec(A)
+  Y = SpaceGerm(X, I)
+  restr_map = SpecMor(Y, X, hom(OO(X), OO(Y), gens(OO(Y)), check=false), check=false)
+  return Y, restr_map
+end
+
+function germ_at_point(A::MPolyRing, a::Vector)
+  X = Spec(A)
+  Y = SpaceGerm(X, a)
+  restr_map = SpecMor(Y, X, hom(OO(X), OO(Y), gens(OO(Y)), check=false), check=false)
+  return Y, restr_map
+end
+
+function germ_at_point(A::MPolyQuo, I::Ideal)
+  X = Spec(A)
+  Y = SpaceGerm(X, I)
+  restr_map = SpecMor(Y, X, hom(OO(X), OO(Y), gens(OO(Y)), check=false), check=false)
+  return Y, restr_map
+end
+
+function germ_at_point(A::MPolyQuo, a::Vector)
+  X = Spec(A)
+  Y = SpaceGerm(X, a)
+  restr_map = SpecMor(Y, X, hom(OO(X), OO(Y), gens(OO(Y)), check=false), check=false)
+  return Y, restr_map
+end
+
+LocalRing = Union{MPolyQuoLocalizedRing{<:Any, <:Any, <:Any, <:Any, 
+                                        <:MPolyComplementOfKPointIdeal},
+                  MPolyLocalizedRing{<:Any, <:Any, <:Any, <:Any, 
+                                     <:MPolyComplementOfKPointIdeal},
+                  MPolyQuoLocalizedRing{<:Any, <:Any, <:Any, <:Any, 
+                                        <:MPolyComplementOfPrimeIdeal},
+                  MPolyLocalizedRing{<:Any, <:Any, <:Any, <:Any, 
+                                     <:MPolyComplementOfPrimeIdeal}
+                 }
+
+function germ_at_point(A::LocalRing; check::Bool=true)
+  X = Spec(A)
+  Y = SpaceGerm(X)
+  restr_map = SpecMor(Y, X, hom(OO(X), OO(Y), gens(OO(Y)), check=false), check=false)
```

Comment:
`X` and `Y` are mathematically the same, only that `Y` knows that it's a germ. Since the constructor is for a local ring, I don't see the point in creating `X` and even returning the identity map.

Reply:
I now created a constructor for the SpaceGerm in this settingonly, but for compatibility I left the germ_at_point for this case and made a comment that it is just for compatibility.